apps/aehttp/src/sc_ws_handler.erl

```diff
@@ -16,38 +16,43 @@
-                               Action =:= <<"shutdown_sign">>;
-                               Action =:= <<"shutdown_sign_ack">>;
-                               Action =:= <<"update">>;
-                               Action =:= <<"update_ack">>).
--define(ACTION_TAG(Action), case Action of
-                                <<"initiator_sign">> -> create_tx;
-                                <<"deposit_tx">> -> deposit_tx;
-                                <<"deposit_ack">> -> deposit_created;
-                                <<"withdraw_tx">> -> withdraw_tx;
-                                <<"withdraw_ack">> -> withdraw_created;
-                                <<"responder_sign">> -> funding_created;
-                                <<"update">> -> update;
-                                <<"update_ack">> -> update_ack;
-                                <<"shutdown_sign">> -> shutdown;
-                                <<"shutdown_sign_ack">> -> shutdown_ack;
-                                <<"leave">> -> leave
+-define(METHOD_SIGNED(Method), Method =:= <<"channels.initiator_sign">>;
+                               Method =:= <<"channels.deposit_tx">>;
+                               Method =:= <<"channels.deposit_ack">>;
+                               Method =:= <<"channels.withdraw_tx">>;
+                               Method =:= <<"channels.withdraw_ack">>;
+                               Method =:= <<"channels.responder_sign">>;
+                               Method =:= <<"channels.shutdown_sign">>;
+                               Method =:= <<"channels.shutdown_sign_ack">>;
+                               Method =:= <<"channels.update">>;
+                               Method =:= <<"channels.update_ack">>).
+-define(METHOD_TAG(Method), case Method of
+                                <<"channels.initiator_sign">>    -> create_tx;
+                                <<"channels.deposit_tx">>        -> deposit_tx;
+                                <<"channels.deposit_ack">>       -> deposit_created;
+                                <<"channels.withdraw_tx">>       -> withdraw_tx;
+                                <<"channels.withdraw_ack">>      -> withdraw_created;
+                                <<"channels.responder_sign">>    -> funding_created;
+                                <<"channels.update">>            -> update;
+                                <<"channels.update_ack">>        -> update_ack;
+                                <<"channels.shutdown_sign">>     -> shutdown;
+                                <<"channels.shutdown_sign_ack">> -> shutdown_ack;
+                                <<"channels.leave">>             -> leave
                             end).
 
 init(Req, _Opts) ->
-    {cowboy_websocket, Req, maps:from_list(cowboy_req:parse_qs(Req))}.
+    lager:debug("init(~p, ~p)", [Req, _Opts]),
+    {cowboy_websocket, Req,
+     maps:merge(#{protocol => <<"legacy">>},
```

Comment:
[cowboy_req:parse_qs returns keys as binary](https://ninenines.eu/docs/en/cowboy/2.2/manual/cowboy_req.parse_qs/), right? How can this protocol atom act as default then? It means that point in code below (only protocol/1 function?) that uses protocol atom as key in map always fetches this item rather than anything from the param.

Reply:
I see answered in https://github.com/aeternity/epoch/pull/1632#issuecomment-428554067